src/pages/TaskQueue.tsx

```diff
@@ -59,41 +61,62 @@ export const TaskQueue = () => {
 
   return (
     <PageWrapper>
-      <H2>Task Queue</H2>
-      <TableControlOuterRow>
-        <SearchableDropdownWrapper>
-          <SearchableDropdown
-            data-cy="distro-dropdown"
-            label="Distro"
-            options={distros}
-            searchFunc={handleSearch}
-            optionRenderer={(option, onClick) => (
-              <DistroOption
-                option={option}
-                key={`distro-select-search-option-${option.id}`}
-                onClick={onClick}
+      {/* @ts-expect-error  */}
+      <StyledH2>Task Queue</StyledH2>
+
+      {selectedDistro === null ? (
+        <Skeleton active />
+      ) : (
+        <>
+          <TableControlOuterRow>
+            <SearchableDropdownWrapper>
+              <SearchableDropdown
+                data-cy="distro-dropdown"
+                label="Distro"
+                options={distros}
+                searchFunc={handleSearch}
+                optionRenderer={(option, onClick) => (
+                  <DistroOption
+                    option={option}
+                    key={`distro-select-search-option-${option.id}`}
+                    onClick={onClick}
+                  />
+                )}
+                onChange={onChangeDistroSelection}
+                value={selectedDistro}
+                buttonRenderer={(option: Partial<TaskQueueDistro>) => (
+                  <DistroLabel>
+                    <StyledBadge>{`${option?.taskCount || 0} ${
+                      option?.taskCount === 1 ? "TASK" : "TASKS"
+                    }`}</StyledBadge>
+                    <StyledBadge>{`${option?.hostCount || 0} ${
+                      option?.hostCount === 1 ? "HOST" : "HOSTS"
+                    }`}</StyledBadge>
+                    <DistroName> {option?.id} </DistroName>
+                  </DistroLabel>
+                )}
               />
-            )}
-            onChange={onChangeDistroSelection}
-            value={selectedDistro}
-            buttonRenderer={(option: Partial<TaskQueueDistro>) => (
-              <DistroLabel>
-                <StyledBadge>{`${option?.taskCount} ${
-                  option?.taskCount === 1 ? "TASK" : "TASKS"
-                }`}</StyledBadge>
-                <StyledBadge>{`${option?.hostCount} ${
-                  option?.hostCount === 1 ? "HOST" : "HOSTS"
-                }`}</StyledBadge>
-                <DistroName> {option?.id} </DistroName>
-              </DistroLabel>
-            )}
-          />
-        </SearchableDropdownWrapper>
-      </TableControlOuterRow>
+            </SearchableDropdownWrapper>
+          </TableControlOuterRow>
+
+          {
+            /* Only show name & link if distro exists. */
+            selectedDistro && (
+              <TableHeader>
+                {/* @ts-expect-error  */}
+                <StyledH3> {selectedDistro.id} </StyledH3>
+                <StyledRouterLink to={`/hosts?distroId=${selectedDistro.id}`}>
```

Comment:
Instead of hard coding this link can you export it as a util function from here. https://github.com/evergreen-ci/spruce/blob/main/src/constants/routes.ts

Also do not hard code query params use a util function to generate them.